nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-frontend/src/main/nifi/src/app/pages/flow-designer/service/canvas-utils.service.ts

```diff
@@ -464,6 +464,28 @@ export class CanvasUtils {
         return selection.size() === 1 && selection.classed('funnel');
     }
 
+    /**
+     * Determines whether the current selection is a stateful processor.
+     *
+     * @param {selection} selection
+     */
+    public isStatefulProcessor(selection: any): boolean {
+        // ensure the correct number of components are selected
+        if (selection.size() !== 1) {
+            return false;
+        }
+        if (this.canRead(selection) === false || this.canModify(selection) === false) {
+            return false;
+        }
+
+        if (this.isProcessor(selection)) {
+            const processorData: any = selection.datum();
+            return processorData.component.persistsState === true;
+        } else {
+            return false;
+        }
+    }
```

Comment:
This name is a bit misleading. It does more than just determines if something is a stateful processor or not. Technically, a process can be stateful even if you can't read/write it.

It probably doesn't matter though. We aren't really using this in any other context.

Reply:
This naming and logic was brought over from NiFi without much thought. I can rename the method if desirable. The logic is correct. We want to verify the selection contains a single Processor, the user has access, and it persists state.

Reply:
This name is fine.